### PR TITLE
Adding ability to run tests against the templates

### PR DESCRIPTION
**Pull request by stack72**

### CODE REVIEW

**Comment by justinvp** [comment]
@stack72, LMK if/when this is ready for review.

**Comment by stack72** [comment]
@justinvp this is ready for review - as we start to add the credentials, we can release the strictness of the conditions. But I will add a followup issue to track the additions of the other templates

File: tests/template_test.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+			stdOut, stdErr := e.RunCommand("pulumi", "new", templateName, "-f", "-s", "template-test")
+
+			path, err := workspace.DetectProjectPathFrom(e.RootPath)
+			assert.NoError(t, err)
+			assert.NotEmpty(t, path)
+
+			if path == "" {
+				panic(fmt.Sprintf("path is empty: %v: %v", stdOut, stdErr))
+			}
```
Comment (justinvp) [inline tests/template_test.go]: Why the `if path == ""` check? Aren't we asserting `path` isn't empty on line 87?

Comment (stack72) [reply tests/template_test.go]: great catch! this was debugging

**Pull request merged**